background/rust.tex

```diff
@@ -312,7 +314,39 @@ \subsubsection{Loops}
 % \label{lst:rust:result}
 % \end{listing}
 
-\include*{background/rust/closures}
+\subsubsection{Closures}
+\label{sec:back:rust:closures}
+
+Closures, a type of anonymous functions, exposes many expressive programming patterns which have emerged the last few years through the popularization of JavaScript \todo{Is this statement true? Maybe its gotten more recognized because of JS, but  closures have been around since at least the 70s...}, and several functional languages.
+The consquence is that both C++ and Java has added support for creating anonymous functions called lambda expressions.
+
+A closure is a function has free variable which is contained within the scope the function is defined.
+The function is said to close over these variables called the environment of the closure.
+The closure can be used as an argument to another function or returned from a maker function.
+\autoref{lst:closure:filter} shows how a closure can be used to filter a list of numbers.
+\todo{Grammar in this paragraph needs to be fixed}
+
+\begin{listing}[H]
+  \begin{minted}{rust}
+fn main() {
+  let (nums, limit) = (vec![1, 2, 3, 4], 3);
```

Comment:
Is, this ideomatic. I think it might be confusing to the user ås this implies that limit and nums are related ås in a compound datatype